### PR TITLE
replaced json with rapidjson

### LINKED ISSUE #317: Switch from python-json to python-rapidjson
[python-rapidjson](https://github.com/kenrobbins/python-rapidjson) is a python wrapper around [rapidjson](https://github.com/miloyip/rapidjson) which is a fully RFC compliant c++ implementation of a JSON parser.

Although the python bindings do not implement all of the api provided by rapidjson it does provide an interface similar to python json which is all that we need at this point.

Some preliminary tests:
```python
import json
import time

import rapidjson

import bigchaindb


def speedtest_serialize_block_json():
    # create a block
    b = bigchaindb.Bigchain()
    tx = b.create_transaction(b.me, b.me, None, 'CREATE')
    tx_signed = b.sign_transaction(tx, b.me_private)
    block = b.create_block([tx_signed] * 1000)

    time_start = time.time()
    for _ in range(1000):
        _ = json.dumps(block, skipkeys=False, ensure_ascii=False, sort_keys=True)
    time_elapsed = time.time() - time_start

    print('speedtest_serialize_block_json: {} s'.format(time_elapsed))


def speedtest_serialize_block_rapidjson():
    # create a block
    b = bigchaindb.Bigchain()
    tx = b.create_transaction(b.me, b.me, None, 'CREATE')
    tx_signed = b.sign_transaction(tx, b.me_private)
    block = b.create_block([tx_signed] * 1000)

    time_start = time.time()
    for _ in range(1000):
        _ = rapidjson.dumps(block, skipkeys=False, ensure_ascii=False, sort_keys=True)
    time_elapsed = time.time() - time_start

    print('speedtest_serialize_block_rapidjson: {} s'.format(time_elapsed))


def speedtest_deserialize_block_json():
    # create a block
    b = bigchaindb.Bigchain()
    tx = b.create_transaction(b.me, b.me, None, 'CREATE')
    tx_signed = b.sign_transaction(tx, b.me_private)
    block = b.create_block([tx_signed] * 1000)
    block_serialized = json.dumps(block, skipkeys=False, ensure_ascii=False, sort_keys=True)

    time_start = time.time()
    for _ in range(1000):
        _ = json.loads(block_serialized)
    time_elapsed = time.time() - time_start

    print('speedtest_deserialize_block_json: {} s'.format(time_elapsed))


def speedtest_deserialize_block_rapidjson():
    # create a block
    b = bigchaindb.Bigchain()
    tx = b.create_transaction(b.me, b.me, None, 'CREATE')
    tx_signed = b.sign_transaction(tx, b.me_private)
    block = b.create_block([tx_signed] * 1000)
    block_serialized = rapidjson.dumps(block, skipkeys=False, ensure_ascii=False, sort_keys=True)

    time_start = time.time()
    for _ in range(1000):
        _ = rapidjson.loads(block_serialized)
    time_elapsed = time.time() - time_start

    print('speedtest_deserialize_block_rapidjson: {} s'.format(time_elapsed))

speedtest_serialize_block_json: 29.55320143699646 s
speedtest_serialize_block_rapidjson: 3.7331912517547607 s
speedtest_deserialize_block_json: 8.043665409088135 s
speedtest_deserialize_block_rapidjson: 7.547366142272949 s
```

Almost 10x increase in serialization time and comparable speed in deserialization.

### PR DESCRIPTION
**Pull request by r-marques**
resolves #317

### CODE REVIEW

File: bigchaindb/util.py [hunk 1]
```diff
@@ -109,8 +110,7 @@ def serialize(data):
         str: JSON formatted string
 
     """
-    return json.dumps(data, skipkeys=False, ensure_ascii=False,
-                      separators=(',', ':'), sort_keys=True)
+    return rapidjson.dumps(data, skipkeys=False, ensure_ascii=False, sort_keys=True)
```
Comment (ttmc) [inline bigchaindb/util.py]: Does `rapidjson.dumps()` assume the equivalent of `separators=(',', ':')` by default?

**Comment by ttmc** [comment]
I guess the documentation about JSON Serialization should be updated.

http://bigchaindb.readthedocs.io/en/latest/json-serialization.html

**Comment by r-marques** [comment]
I will do that

File: bigchaindb/util.py (hunk 1 above)
Comment (r-marques) [inline bigchaindb/util.py]: At least the python wrapper does not provide that argument but it uses `(',', ':')` by default

**Pull request merged**